tests/integration/netapi/rest_cherrypy/app_pam_test.py

```diff
@@ -0,0 +1,141 @@
+# coding: utf-8
+'''
+Integration Tests for restcherry salt-api with pam eauth
+'''
+
+# Import python libs
+from __future__ import absolute_import
+import os
+
+# Import salttesting libs
+from salttesting.unit import skipIf
+from salttesting.helpers import (
+    ensure_in_syspath,
+    destructiveTest)
+ensure_in_syspath('../../../')
+
+from tests.utils import BaseRestCherryPyTest
+
+# Import Salt Libs
+import integration
+import salt.utils
+
+# Import 3rd-party libs
+# pylint: disable=import-error,unused-import
+from salt.ext.six.moves.urllib.parse import urlencode  # pylint: disable=no-name-in-module
+try:
+    import cherrypy
+    HAS_CHERRYPY = True
+except ImportError:
+    HAS_CHERRYPY = False
+# pylint: enable=import-error,unused-import
+
+USERA = 'saltdev'
+USERA_PWD = 'saltdev'
+SET_USERA_PWD = '$6$SALTsalt$ZZFD90fKFWq8AGmmX0L3uBtS9fXL62SrTk5zcnQ6EkD6zoiM3kB88G1Zvs0xm/gZ7WXJRs5nsTBybUvGSqZkT.'
+
+AUTH_CREDS = {
+    'username': USERA,
+    'password': USERA_PWD,
+    'eauth': 'pam'}
+
+
+@skipIf(HAS_CHERRYPY is False, 'CherryPy not installed')
+class TestAuthPAM(BaseRestCherryPyTest, integration.ModuleCase):
+    '''
+    Test auth with pam using salt-api
+    '''
+
+    @destructiveTest
+    @skipIf(os.geteuid() != 0, 'You must be logged in as root to run this test')
+    def setUp(self):
+        super(TestAuthPAM, self).setUp()
```

Comment:
@cachedout : Wanted to ensure using super() is this instance is correct, because BaseRestCherryPyTest has a setup() and teardown() method. If I did not call the setup() method using super it would not work. Not sure if there is a better way to approach this. Same thing for the tearDown method.